File: DecisionMaker.py
from adafruit_circuitplayground import cp
import time
import random

#Define colors
pink = (12,10,12)
gold = (50, 40, 5)
blue = (0,0,8)
orange = (25, 10, 0)
blank = (0,0,0)
grn = (0,20,0)
green  = (0,40,10)
red = (20,0,0)
white = (20,20,20)

color = [red,orange,gold,green,blue,white,pink]
statecolor = [red,gold,green,blank]
answers = ["Yes, definitely.", "As I see it, yes.", "Reply hazy, try again.", "Cannot predict now.", "Do not count on it.", "My sources say no.", "Outlook not so good.", "Very doubtful."]
yes = ["m8/yesdef.wav", "m8/asyes.wav"]
maybe = ["m8/hazy.wav", "m8/cannot.wav", "m8/count.wav"]
no = ["m8/sources.wav", "m8/outlook.wav", "m8/doubtful.wav"]

colors = [green,green,orange,orange,orange,red,red,red]

wavs = ["m8/yesdef.wav", "m8/asyes.wav", "m8/hazy.wav", "m8/cannot.wav", "m8/count.wav", "m8/sources.wav", "m8/outlook.wav", "m8/doubtful.wav"]


def cycle(x):
    for i in range(x*10):
        cp.pixels[i%10] = random.choice(color)
        time.sleep(.1)
def dom8():
    cycle(1)
    ans = random.randrange(8)
    cp.pixels.fill(colors[ans])
    print(answers[ans])
    cp.play_file(wavs[ans])
    time.sleep(2)
    cp.pixels.fill(blank)
def blinknum(num,color):
    if num != 0:
        for i in range(num):
            cp.pixels.fill(color)
            time.sleep(.25)
            cp.pixels.fill(blank)
            time.sleep(.10)
    else:
        for i in range(10):
            cp.pixels[i] = color
            cp.pixels.show()
            time.sleep(.14)

        cp.pixels.fill(blank)

magic = 0
brain = 1

state = magic
cycle(2)
cp.pixels.fill(blank)
good = 0
bad = 0
garray = [4,3,2,1,0]
barray = [5,6,7,8,9]



while True:
    if cp.switch:
        if state == magic:
            state = brain
            good = 0
            bad = 0
            g=0
            b=0
            cycle(1)
            cp.pixels.fill(blank)
    else:
        if state==brain:
            state = magic
            cycle(1)
            cp.pixels.fill(blank)

    if cp.button_a and state == magic:
        dom8()

    if cp.button_a and state == brain:
        time.sleep(1)
        g = g+1

        if g>5:
            g = 5
        print (str(g)+":"+str(garray[g-1]))

        cp.pixels[garray[g-1]] = green

    if cp.button_b and state == brain:
        time.sleep(1)
        b = b+1
        if b>5:
            b = 5

        cp.pixels[barray[b-1]] = red

    if cp.touch_A1 and state==brain:
        good = good+g
        bad = bad+b
        g=0
        b=0

    if cp.touch_A2:
        cycle(1)
        ans = 0
        if good==bad:
            ans = 1
        if good > bad:
            ans = 0
        if bad > good:
            ans = 2
        good = 0
        bad = 0
        g = 0
        b = 0
        if ans == 0:
            cp.pixels.fill(green)
            cp.play_file(random.choice(yes))
        if ans == 1:
            cp.pixels.fill(gold)
            cp.play_file(random.choice(maybe))
        if ans == 2:
            cp.pixels.fill(red)
            cp.play_file(random.choice(no))
        time.sleep(2)
        cp.pixels.fill(blank)

File: DecisionMaker.js
input.onSwitchMoved(SwitchDirection.Right, function () {
    light.showAnimation(light.rainbowAnimation, 500)
    light.setAll(0x000000)
})
function cleartally() {
    bad = 0
    good = 0
}
function magic8() {
    light.showAnimation(light.rainbowAnimation, 500)
    light.setAll(0x000000)
    ans = Math.randomRange(0, 7)
    light.setAll(colors[ans])
    pause(2000)
    light.setAll(0x000000)
}
function magic8text() {
    light.showAnimation(light.rainbowAnimation, 500)
    light.setAll(0x000000)
    ans = Math.randomRange(0, 7)
    light.setAll(colors[ans])
    keyboard.type("" + answers[ans])
    pause(2000)
    light.setAll(0x000000)
}
input.onSwitchMoved(SwitchDirection.Left, function () {
    light.showAnimation(light.rainbowAnimation, 500)
    light.setAll(0x000000)
    cleartally()
    g = 0
    b = 0
    savans = 0
})
input.touchA7.onEvent(ButtonEvent.Click, function () {
    if (!(input.switchRight()) && savans != 0) {
        light.showAnimation(light.rainbowAnimation, 500)
        if (2 == savans) {
            light.setAll(0xffff00)
        } else if (3 == savans) {
            light.setAll(0xff0000)
        } else {
            light.setAll(0x00ff00)
        }
    }
    pause(2000)
    light.setAll(0x000000)
})
input.buttonB.onEvent(ButtonEvent.Click, function () {
    if (!(input.switchRight())) {
        b += 1
        if (b > 5) {
            b = 5
        }
        light.setPixelColor(barray[b - 1], 0xff0000)
    }
})
input.buttonA.onEvent(ButtonEvent.Click, function () {
    if (input.switchRight()) {
        magic8()
    } else {
        g += 1
        if (g > 5) {
            g = 5
        }
        light.setPixelColor(garray[g - 1], 0x00ff00)
    }
})
input.touchA4.onEvent(ButtonEvent.Click, function () {
    if (input.switchRight()) {
        magic8text()
    }
})
input.touchA1.onEvent(ButtonEvent.Click, function () {
    if (!(input.switchRight())) {
        good += g
        bad += b
        b = 0
        g = 0
        light.setAll(0x000000)
    }
})
input.touchA2.onEvent(ButtonEvent.Click, function () {
    b = 0
    g = 0
    light.showAnimation(light.rainbowAnimation, 500)
    if (!(input.switchRight())) {
        if (good == bad) {
            light.setAll(0xffff00)
            savans = 2
        } else if (good > bad) {
            light.setAll(0x00ff00)
            savans = 1
        } else {
            light.setAll(0xff0000)
            savans = 3
        }
    }
    pause(2000)
    light.setAll(0x000000)
    cleartally()
})
let savans = 0
let b = 0
let g = 0
let ans = 0
let colors: number[] = []
let answers: string[] = []
let barray: number[] = []
let garray: number[] = []
let bad = 0
let good = 0
good = 0
bad = 0
light.showAnimation(light.rainbowAnimation, 500)
light.setAll(0x000000)
garray = [4, 3, 2, 1, 0]
barray = [5, 6, 7, 8, 9]
answers = ["Yes, definitely.", "As I see it, yes.", "Reply hazy, try again.", "Cannot predict now.", "Do not count on it.", "My sources say no.", "Outlook not so good.", "Very doubtful."]
colors = [Colors.Green, Colors.Green, Colors.Yellow, Colors.Yellow, Colors.Yellow, Colors.Red, Colors.Red]
